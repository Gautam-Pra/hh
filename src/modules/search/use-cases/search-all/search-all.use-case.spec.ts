import { beforeAll, describe, expect, test } from 'vitest'
import { SearchAllUseCase } from '#modules/search/use-cases'

describe('SearchEverything', () => {
  let searchAllUseCase: SearchAllUseCase

  beforeAll(() => {
    searchAllUseCase = new SearchAllUseCase()
  })

  test('should search everything by query and return a list of results', async () => {
    const results = await searchAllUseCase.execute('imagine+dragons+bones')

    expect(results).toMatchSnapshot()
  })
})
