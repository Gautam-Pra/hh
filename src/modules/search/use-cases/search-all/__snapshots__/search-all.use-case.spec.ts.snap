// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchEverything > should search everything by query and return a list of results 1`] = `
{
  "albums": {
    "position": 2,
    "results": [
      {
        "artist": undefined,
        "description": "2021 · English Album · Arcane, League of Legends",
        "id": "30690651",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/273/Arcane-League-of-Legends-Soundtrack-from-the-Animated-Series--English-2021-20211117001130-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/273/Arcane-League-of-Legends-Soundtrack-from-the-Animated-Series--English-2021-20211117001130-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/273/Arcane-League-of-Legends-Soundtrack-from-the-Animated-Series--English-2021-20211117001130-500x500.jpg",
          },
        ],
        "language": "english",
        "position": undefined,
        "songIds": "iShuYW5v, NihjV1ye, TiIwRuAy, -nAVuln8, NLIW3rG4, Iw6Mkj-M, zUw-eAdj, qv4j6PQc, 5ZhrtDN_, eGGb1baQ, NilyG0mP",
        "title": "Arcane League of Legends (Soundtrack from the Animated Series)",
        "type": "album",
        "url": undefined,
        "year": "2021",
      },
      {
        "artist": undefined,
        "description": "2015 · English Album · Various Artists",
        "id": "1658369",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/057/Acoustix-3-English-2015-20240113102054-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/057/Acoustix-3-English-2015-20240113102054-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/057/Acoustix-3-English-2015-20240113102054-500x500.jpg",
          },
        ],
        "language": "english",
        "position": undefined,
        "songIds": "3PyY4aPI, DQV8CfqJ, NHGS0oI8, g5GsYjYk, wNDbhlNp, 3jNyKHsc, Lava9XJs, 1i275ZSo, PttOq5Qw, q11IfWw0, 3-NVzq83, 6ug_NUMY, Vhzva1im, k1V2yhZr, mQkNT-oY, obCW2akb, JjKKyOOK, 08WHil7_, 2ItHNmf8, sS6ytDnx, LXJwB5Qw, sUKNzQPT, r_1aZS3R, jrpkPst7, _ovxcmOZ, cLxVe5jm, gpG6GIGa, FSdITuDG, yb6FPDYo, pJHpTgCS, A__yq_B6, F-E2Vlri, VWZFpMFp, 77p_aV3N, CkhcG8ai, 0NdMvvK4, 2Ix-WNRO, 7l3pOQnQ, 9Oq75l3O, KRpLJ2KF",
        "title": "Acoustix 3",
        "type": "album",
        "url": undefined,
        "year": "2015",
      },
      {
        "artist": undefined,
        "description": "2022 · English Album · Various Artists",
        "id": "38112005",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/794/Rainy-Days-2022-English-2022-20230207223116-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/794/Rainy-Days-2022-English-2022-20230207223116-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/794/Rainy-Days-2022-English-2022-20230207223116-500x500.jpg",
          },
        ],
        "language": "english",
        "position": undefined,
        "songIds": "iXoaHVC3, p0f1i-cG, g8WlADy9, mnEenbK-, RLjTAuOP, cR_r-Mtz, iaFkiupr, c23kGYzL, n_w-yiYu, TiCELM52, Ri34c2J9, 5Cm_i6C-, pjGR3zm_, r19sc3zS, Qg6C0wak, 5IkXN3A7, O5qliq6s, bCa6JdBy, dZ5gVtQl, 5wxbS7T7, XTZ3CHbJ, qH6p40b4, k0Zmhv0n, py7hcbAW, yAteOSkm, 0RfMXVwc, zOrSZ0Tq, aM4_4q-_, MHWLSAxQ, 07AoRrzd, wu1xxLdY, uMPYVXhw, n6ty30Qf, PUADRZ-X, b1UniD8D, l0ENEPHe, gJ6EZSBf, KXvxpFcP, 1aNy4xWQ, hNUnsuzn, _VAz2ODN, q_fDiuFj, sT-Ch9T_, plqLuz2B, fLku5opv, AZ6qnuES, znSnT9yx, ymmwvt4W, DZlNTMS_, CVAWCRSK, PXWBdZas, I8bfh5_b, T_Pq6Nuk, XRLrjsHV, 19JMYl5D, RslhheE-, A9fT4WT_, TphPkJYI, 32C-ynoC, ozYfUtLk, LKBoj9LG, J8JVS9JQ, Z-AuVaGk, ViZHJc9t, Mx8ofMNL, BDnHp8gz, wTlTf7Vi, chbwwaeq, jWb5w_aA, koB5EWON, j5V5hQKF, ojqt9X2Y, X0sP3dO2, u-__KIHi, e78yR_Rz, wbhMY8OV, ztundUK-, X-eqPEak, eRCzMakl, iKVBwhlU, br10EetC",
        "title": "Rainy Days 2022",
        "type": "album",
        "url": undefined,
        "year": "2022",
      },
    ],
  },
  "artists": {
    "position": 4,
    "results": [],
  },
  "playlists": {
    "position": 3,
    "results": [],
  },
  "songs": {
    "position": 1,
    "results": [
      {
        "album": undefined,
        "description": "Hit The Button Karaoke · Karaoke Picks, Vol. 115",
        "id": "hMs277rM",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/648/Karaoke-Picks-Vol-115-English-2022-20220330150959-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/648/Karaoke-Picks-Vol-115-English-2022-20220330150959-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/648/Karaoke-Picks-Vol-115-English-2022-20220330150959-500x500.jpg",
          },
        ],
        "language": "english",
        "position": undefined,
        "primaryArtists": "Hit The Button Karaoke",
        "singers": "Hit The Button Karaoke",
        "title": "Bones (Originally Performed by Imagine Dragons) (Instrumental Version)",
        "type": "song",
        "url": undefined,
      },
      {
        "album": undefined,
        "description": "Party Tyme Karaoke · Party Tyme - Pop Male Hits 12 (Karaoke Versions)",
        "id": "wfelL3hY",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/105/Party-Tyme-Pop-Male-Hits-12-Karaoke-Versions-English-2022-20221124053921-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/105/Party-Tyme-Pop-Male-Hits-12-Karaoke-Versions-English-2022-20221124053921-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/105/Party-Tyme-Pop-Male-Hits-12-Karaoke-Versions-English-2022-20221124053921-500x500.jpg",
          },
        ],
        "language": "english",
        "position": undefined,
        "primaryArtists": "Party Tyme Karaoke",
        "singers": "Party Tyme Karaoke",
        "title": "Bones (Made Popular By Imagine Dragons) [Karaoke Version]",
        "type": "song",
        "url": undefined,
      },
      {
        "album": undefined,
        "description": "Hit The Button Karaoke · Karaoke Picks, Vol. 115",
        "id": "Du3u2_MT",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/648/Karaoke-Picks-Vol-115-English-2022-20220330150959-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/648/Karaoke-Picks-Vol-115-English-2022-20220330150959-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/648/Karaoke-Picks-Vol-115-English-2022-20220330150959-500x500.jpg",
          },
        ],
        "language": "english",
        "position": undefined,
        "primaryArtists": "Hit The Button Karaoke",
        "singers": "Hit The Button Karaoke",
        "title": "Bones (Originally Performed by Imagine Dragons) (Karaoke Version)",
        "type": "song",
        "url": undefined,
      },
    ],
  },
  "topQuery": {
    "position": 0,
    "results": [],
  },
}
`;
